clang/include/clang/Basic/DiagnosticGroups.td

```diff
@@ -275,6 +275,9 @@ def CPre2xCompat : DiagGroup<"pre-c2x-compat">;
 def CPre2xCompatPedantic : DiagGroup<"pre-c2x-compat-pedantic",
                                      [CPre2xCompat]>;
 
+// Warning for treating C input as C++ input.
+def ExpectedFileType : DiagGroup<"ExpectedFileType">;
```

Comment:
The name of the diagnostic when used as an option probably should be something like:  `DiagGroup<"expected-file-type">`

Reply:
Yes, I have verified the other group naming. I will update this.